tenpy/algorithms/mps_common.py

```diff
@@ -2392,3 +2393,89 @@ def update_local(self, _, optimize=True):
+        and we need to disregard those eigenvectors of the larger one, that have eigenvalue zero,
+        since we dont have corresponding eigenvalues of the smaller one.
+
+    Options
+    -------
+    .. cfg:config :: QRBasedVariationalApplyMPO
+        :include: VariationalApplyMPO
+
+        cbe_expand : float
+            Expansion rate. The QR-based decomposition is carried out at an expanded bond dimension
+            ``eta = (1 + cbe_expand) * chi``, where ``chi`` is the bond dimension before the time step.
+            Default is `0.1`.
+        cbe_expand_0 : float
+            Expansion rate at low ``chi``.
+            If given, the expansion rate decreases linearly from ``cbe_expand_0`` at ``chi == 1``
+            to ``cbe_expand`` at ``chi == trunc_params['chi_max']``, then remains constant.
+            If not given, the expansion rate is ``cbe_expand`` at all ``chi``.
+        cbe_min_block_increase : int
+            Minimum bond dimension increase for each block. Default is `1`.
+        use_eig_based_svd : bool
+            Whether the SVD of the bond matrix :math:`\Xi` should be carried out numerically via
+            the eigensystem. This is faster on GPUs, but less accurate.
+            It makes no sense to do this on CPU. It is currently not supported for update_imag.
+            Default is `False`.
+        compute_err : bool
+            Whether the truncation error should be computed exactly.
+            Compared to SVD-based TEBD, computing the truncation error is significantly more expensive.
+            If `True` (default), the full error is computed.
+            Otherwise, the truncation error is set to NaN.
+    """
+
+    def _expansion_rate(self, i):
+        """get expansion rate for updating bond i"""
+        expand = self.options.get('cbe_expand', 0.1)
+        expand_0 = self.options.get('cbe_expand_0', None)
+
+        if expand_0 is None or expand_0 == expand:
+            return expand
+
+        chi_max = self.trunc_params.get('chi_max', None)
+        if chi_max is None:
+            raise ValueError('Need to specify trunc_params["chi_max"] in order to use cbe_expand_0.')
+
+        chi = min(self.psi.get_SL(i).shape)
+        return max(expand_0 - chi / chi_max * (expand_0 - expand), expand)
+
+    def update_new_psi(self, theta: npc.Array):
+        """Given a new two-site wave function `theta`, split it and save it in :attr:`psi`."""
+        i0 = self.i0
+        new_psi = self.psi
+        # old_A0 = new_psi.get_B(i0, form='A')
+        expand = self._expansion_rate(i0)
+        U, S, VH, err, renormalize = qr_theta(self.psi.get_B(i0, 'B'), self.psi.get_B(i0+1, 'B'), theta,
+                                              self.trunc_params, 
+                                              expand=expand, 
+                                              use_eig_based_svd=self.options.get('use_eig_based_svd', False),
```

Comment:
As implemented now, using ``A S B`` form, we can not support eig based svd.

I am not sure if there is a way to get around this, similar to the inversion free trick in TEBD.
If not, we should remove the ``use_eig_based_svd`` option.

If yes, we should implement that.

Reply:
``QRBasedVariationalApplyMPO`` now supports ``use_eig_based_svd`` by using ``t_L S t_R`` tensors, where ``t_L`` and ``t_R`` are in ``'A'``, ``'B'`` or ``'Th'`` form depending on the input parameters. ``VariationalApplyMPO`` already supports the possibility to update ``new_psi`` with tensors in ``'Th'`` form.